### ISSUE TITLE
Add video progress bar (from Aerial Dream) 

### ISSUE THREAD

**Issue by youngkong10** [theothernt/AerialViews#146]
Would love to have the video progress bar from aerial dream back, also I was wondering which font settings are closest to the original fonts/sizing from aerial dream? Thanks for carrying on that projects functionality by the way I love it.

**Comment by theothernt** [comment]
It's been [requested a few times](https://github.com/theothernt/AerialViews/issues/21) so I hope to add the feature in the next two releases - so maybe 1.7.2 or 1.7.3.

As for the text settings in Aerial Dream, try changing to font to Roboto and the text size to 24sp.

**Comment by youngkong10** [comment]
Oh sorry I didn't see that, thanks for letting me know. Also thanks for the font suggestion it looks nice.

**Comment by theothernt** [comment]
No problem at all. As you've moved over from Aerial Dream, I'd suggest taking 5 or 10 minutes to go through all of the options!

**Comment by theothernt** [comment]
It's likely the progress bar will return in v1.7.4.

I have done some initial work on the feature but it turned out I have to tidy up the Image and Video player a bit first.

Also, you'll be able to place the progress bar at the top or bottom of the screen and adjust the opacity!

**Comment by youngkong10** [comment]
Oh cool! Thanks for updating this thread, and thanks for all the development work on the app too.

**Comment by theothernt** [comment]
Do you remember, did the progress bar in Aerial Dream move smoothly across the screen or did it appear to jump with each update?

**Comment by youngkong10** [comment]
I remember the progress bar being smooth in Aerial Dream, almost as though it was just a separate progress animation that used the length of the current video as a reference. Is that what you mean? Or do you mean jumping with each 'update' as in when the current video changes?

**Comment by theothernt** [comment]
I meant jumping with each update, but after re-reading how to old code works - it would have been smooth.

I'm just using a different method this time, so I want to make sure it's the same or better :)

**Comment by youngkong10** [comment]
I'm excited to see it, hope the coding goes smoothly.

**Comment by theothernt** [comment]
Does anyone want to test the progress bar feature with a beta?

**Comment by youngkong10** [comment]
Yes I'd be happy to!

**Comment by theothernt** [comment]
Ok, I have two betas...

1. If you have the APK installed (not the version from Google Play), [this will update it to v1.7.4 beta1](https://www.dropbox.com/scl/fi/3rmba7bx2nulnwo1dh31a/aerial-views-1.7.4-beta1.apk?rlkey=vkgtph71a6z7ka82fv1t70zoz&dl=0)
2. If you have the Google Play version installed or want a 2nd version to test, [this is a debug version of v1.7.4 beta1](https://www.dropbox.com/scl/fi/sinrbz30y61vh0zygeosd/aerial-views-1.7.4-debug-beta1.apk?rlkey=vx184q83ulk1kw2a19gaack94&dl=0)

**Comment by youngkong10** [comment]
Thanks I just set up the second one you sent, everything's working great so far and I'll update if there are any issues. I'll probably try to redo the default screensaver ADB commands later.

**Comment by theothernt** [comment]
Ok, just be aware that for the debug version, the name is different...

`settings put secure screensaver_components com.neilturner.aerialviews.debug/.ui.screensaver.DreamActivity`

**Comment by youngkong10** [comment]
Gotcha thank you for letting me know. This seems to work very well so far. I'm glad there is an opacity setting and everything seems stable.

**Comment by youngkong10** [comment]
Hm, the adb command method doesn't seem to work for me now. Never had a problem setting it before, and I made sure it's the correct debug naming. I tried setting it to default and then boot cycling and then doing the adb command again but no luck. Any ideas?

**Comment by theothernt** [comment]
Is issue with ADB not working properly or are you saying the screensaver isn't starting like before?

**Comment by youngkong10** [comment]
Yeah sorry what I meant was I'm seeing the "correct" thing for   settings get secure screensaver_components   but it's still just going to default screensaver.

**Comment by theothernt** [comment]
Just to rule out something - can you check in Settings > About and make sure it says **beta.debug** ?

**Comment by youngkong10** [comment]
Yeah of course. It does say beta.debug

**Comment by theothernt** [comment]
Ok, this should work...

`adb shell settings put secure screensaver_components "com.neilturner.aerialviews.debug/com.neilturner.aerialviews.ui.screensaver.DreamActivity"
`

**Comment by youngkong10** [comment]
That works great thank you! Still no issues so far.

**Comment by theothernt** [comment]
v1.7.4 has been released on GitHub (APK), the Play Store version should be available in a week or two.

**Comment by theothernt** [comment]
I haven't seen that issue but it could be related to the video type. Is it Dolby Vision, is the frame rate of the video different?

Also, have you changed the opacity of the progress bar?

**Comment by theothernt** [comment]
Thanks for the video, I've never seen an issue like that. What device or TV are you using?

If you put the progress bar at the top of the screen, does the issue still appear?

Also, if you're using AV from the Play Store, you can opt-in to the beta which has a few fixes?

**Comment by theothernt** [comment]
If you like, I can make a beta APK for you - I could move the progress bar up a bit to test if that fixes the issue?

**Comment by theothernt** [comment]
This won't be a lot of work. I'll add some padding to the bottom, but a little more to the top... so we can see if there's a difference...

[Aerial Views beta build](https://www.dropbox.com/scl/fi/h2qun5syu7j0hnhb4al4o/aerial-views-1.7.6-beta3.apk?rlkey=sv91h3uiutcspnz3x15oac2tg&dl=0)

**Comment by theothernt** [comment]
When you say 100% do you mean the opacity or a bug where the progress doesn't move?

**Comment by theothernt** [comment]
Thanks, I'll look into fixing that ASAP and report back with a new beta build.

**Comment by theothernt** [comment]
If the issue persists, it might be best to open a new issue.

**Comment by theothernt** [comment]
> > Just to add, I have added some code to not allow certain bad values (ie. a progress over 100%) but currently, after a video or photos finishes, the progress bar is reset. Let me know if you continue to see the issue.
> 
> How can I test it ? Do you have a build for it ? Many thanks in advance

That change was in the last beta I have you, but it's also in the just-released v1.7.6 APK here on GitHub.

**Comment by theothernt** [comment]
You can try resetting the app, but I'm not sure that will do anything.

This might not be possible, but testing with a) a different TV and b) a different Android TV device would rule out these two devices. It's just odd that I'm not seeing the issues from anyone else (yet?).

There have been bugs reported that I can't reproduce, so it's not unique on Android.

As for the Progress Bar itself, it's just a view (think of a box). I give it a background colour, then animate it's size over a period of time. For a photo of 30s, the size goes from 0 to 1920 (on a 1080p TV). It does always start at 0 though.

And displaying a View or TextBlock (Overlays in my app) on top of video or a photo, is an ordinary thing.

**Comment by theothernt** [comment]
Ok, thanks for all the testing. It's an odd issue, hopefully we can track it down!
